### PR TITLE
feat(FX-4189): Add the ability to select an image from the library for reverse image search

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves [FX-4189]

### Description
Add the ability to select an image on the user's device from the library for image search

#### Android
https://user-images.githubusercontent.com/3513494/186924100-89a32f85-efc2-4259-8d77-696ee657437e.mp4

#### iOS
https://user-images.githubusercontent.com/3513494/186925291-e118513a-1df1-45ec-80d6-6624d6b49203.MP4

#### Acceptance Criteria

- The user can select an image on the device from the library

#### Setup Instructions

-

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add the ability to select an image from the library for reverse image search - dimatretyak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[FX-4189]: https://artsyproduct.atlassian.net/browse/FX-4189?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/ReverseImage/Screens/Preview/ReverseImagePreview.tsx [hunk 1]
```diff
@@ -83,9 +87,12 @@ export const ReverseImagePreviewScreen: React.FC<Props> = (props) => {
...
 
-        <CameraFramesContainer />
-
-        <Background height={CAMERA_BUTTONS_HEIGHT} />
+        {!photo.fromLibrary && (
+          <>
+            <CameraFramesContainer />
+            <Background height={CAMERA_BUTTONS_HEIGHT} />
+          </>
+        )}
```
Comment (dimatretyak) [inline src/app/Scenes/ReverseImage/Screens/Preview/ReverseImagePreview.tsx:95]: **question**: should we display frames if the user has selected an image from the library?

| With Frame | Without Frame |
| ------------- | ------------- |
| with-frame | no-frame |

Comment (dblandin) [reply src/app/Scenes/ReverseImage/Screens/Preview/ReverseImagePreview.tsx:95]: Sounds reasonable to remove the frames in that case if it's easy to do so!

**Review by dblandin — APPROVED** [review]
Awesome 💯

**Review by gkartalis — APPROVED** [review]
🔥

**Review by anandaroop — COMMENTED** [review]
🚀

**Pull request merged**